crates/ra_hir_ty/src/infer/path.rs

```diff
@@ -81,7 +81,7 @@ impl<'a> InferenceContext<'a> {
                 let generics = crate::utils::generics(self.db.upcast(), impl_id.into());
                 let substs = Substs::type_params_for_generics(&generics);
```

Comment:
How can substs be not empty here?

Reply:
These are the type parameters of the impl:
```rust
impl<T> SomeType<T> {}
```